### ISSUE TITLE
[DEFECTS EPIC] implement filtering and sorting of the data in the defects table

### ISSUE THREAD

**Issue by ITsvetkoFF** [ITsvetkoFF/Kv-012#82]
As always there is the one simple question with filtering and sorting: 
- will we have all the data at once in the client?
if the answer is "Yes" so filtering is trivial

blocked by #72

**Comment by vp9920** [comment]
Can this task be unblocked now?

**Comment by ITsvetkoFF** [comment]
we shall make the test in our tables (generating as much data as we can to hang Angular)
and then decide if this limit is unreachable for us. 
And if it is. do sorting by means of Angular.
If not - by means of MongoDB

So this task is blocked by such an exploration. 

By the way, we can combine server and client sorting:
For instance, in case of 1000+ items in dataset use server-side sorting.

@vp9920 by the way, can you explore the limits of your smart filters? can it manage 100-500-1000 runs? interesting! (you can create a separate task for that)
